packages/core/src/external/aws/lambda-logic/bundle-to-brief.ts

```diff
@@ -1,128 +1,746 @@
-import { Bundle } from "@medplum/fhirtypes";
-import { errorToString } from "@metriport/shared";
+import * as dotenv from "dotenv";
+dotenv.config();
+// keep that ^ on top
+import { PromptTemplate } from "@langchain/core/prompts";
+import { RecursiveCharacterTextSplitter } from "@langchain/textsplitters";
+import {
+  Address,
+  Bundle,
+  CodeableConcept,
+  Condition,
+  EncounterDiagnosis,
+  HumanName,
+  Location,
+  Medication,
+  Observation,
+  Organization,
+  Patient,
+  Quantity,
+  Resource,
+} from "@medplum/fhirtypes";
+import { toArray } from "@metriport/shared";
+import { ISO_DATE } from "@metriport/shared/common/date";
+import dayjs from "dayjs";
+import { LLMChain, MapReduceDocumentsChain, StuffDocumentsChain } from "langchain/chains";
 import { cloneDeep } from "lodash";
-import { capture, out } from "../../../util";
-import { base64ToString } from "../../../util/base64";
-import { findDiagnosticReportResources, findPatientResource } from "../../fhir/shared";
-import { BedrockUtils } from "../bedrock";
+import { filterBundleByDate } from "../../../command/consolidated/consolidated-filter-by-date";
+import { findPatientResource } from "../../../external/fhir/shared/index";
+import { BedrockChat } from "../../langchain/bedrock/index";
 
-const MAXIMUM_BRIEF_STRING_LENGTH = 500_000;
+const CHUNK_SIZE = 100000;
+const CHUNK_OVERLAP = 1000;
+const relevantResources = [
+  "AllergyIntolerance",
+  "DiagnosticReport",
+  "Immunization",
+  "Location",
+  "Procedure",
+  "Medication",
+  "MedicationAdministration",
+  "MedicationRequest",
+  "MedicationStatement",
+  "Condition",
+  "Observation",
+  "Practitioner",
+  "Organization",
+];
 
-const relevantResources = ["DiagnosticReport", "Patient"];
+const referenceResources = [
+  "Practitioner",
+  "Organization",
+  "Observation", // maybe worth keeping..
+  "Location",
+];
 
 export type Brief = {
   id: string;
   content: string;
   link: string;
 };
 
-function getBedrockUtilsInstance(): BedrockUtils {
-  return new BedrockUtils();
+//--------------------------------
+// AI-based brief generation
+//--------------------------------
+export async function summarizeFilteredBundleWithAI(
+  bundle: Bundle<Resource>
+): Promise<string | undefined> {
+  // filter out historical data
+  const numHistoricalYears = 2;
+  const dateFrom = dayjs().subtract(numHistoricalYears, "year").format(ISO_DATE);
+  const filteredBundle = filterBundleByDate(bundle, dateFrom);
+
+  const inputString = prepareBundleForBrief(filteredBundle);
+
+  // TODO: experiment with different splitters
+  const textSplitter = new RecursiveCharacterTextSplitter({
+    chunkSize: CHUNK_SIZE,
+    chunkOverlap: CHUNK_OVERLAP,
+  });
+  const docs = await textSplitter.createDocuments([inputString ?? ""]);
+
+  const llmSummary = new BedrockChat({
```

Comment:
these should probably not be hardcoded...

Reply:
FYI: you can tie your comment to multiple lines so they are displayed along your comment.

ditto: just reverting a revert, so likely to be addressed on follow-up PRs?

Reply:
not applicable any longer